pytest_inmanta/parameters.py

```diff
@@ -0,0 +1,87 @@
+"""
+    Copyright 2022 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+from inmanta.module import InstallMode
+
+from .test_parameter import (
+    BooleanTestParameter,
+    EnumTestParameter,
+    ListTestParameter,
+    PathTestParameter,
+)
+
+param_group = "pytest-inmanta"
+
+inm_venv = PathTestParameter(
+    argument="--venv",
+    environment_variable="INMANTA_TEST_ENV",
+    usage=(
+        "Folder in which to place the virtual env for tests (will be shared by all tests). "
+        "This options depends on symlink support. This does not work on all windows versions. "
+        "On windows 10 you need to run pytest in an admin shell. "
+        "Using a fixed virtual environment can speed up running the tests."
+    ),
+    group=param_group,
+)
+
+inm_mod_in_place = BooleanTestParameter(
+    argument="--use-module-in-place",
+    environment_variable="INMANTA_USE_MODULE_IN_PLACE",
+    usage=(
+        "Tell pytest-inmanta to run with the module in place, useful for debugging. "
+        "Makes inmanta add the parent directory of your module directory to it's directory path, instead of copying your "
+        "module to a temporary libs directory. "
+        "It allows testing the current module against specific versions of dependent modules. "
+        "Using this option can speed up the tests, because the module dependencies are not downloaded multiple times."
+    ),
+    group=param_group,
+)
+
+inm_mod_repo = ListTestParameter(
+    argument="--module_repo",
```

Comment:
I considered it, and didn't do it because it would just make more deprecation, and I thought people didn't really care about the "-" vs "_".  But given that you seem to prefer "-" as well, it might be worth changing it after all, I will.

Reply:
I generally prefer `-`, but mainly I think consistency is important, and we have a mix of both now.

Reply:
Looks good, but don't we check the legacy options twice now? We check them via the new option's `legacy` field but the constructor also adds them to the registry.

Reply:
The option is registered indeed, otherwise it won't be usable, and someone could call its resolve method, but this is not what the plugins of `pytest-inmanta`, they always use the most recent option, which falls back to the legacy option (eventually)

Reply:
I misinterpreted this for a bit.